integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -764,23 +782,26 @@ fn test_resharding_v3_base(params: TestReshardingParameters) {
         // Give enough time to produce ~7 epochs.
         Duration::seconds((7 * params.epoch_length) as i64),
     );
-    let client = &env.test_loop.data.get(&client_handles[0]).client;
+    let client = &env.test_loop.data.get(&client_handles[client_index]).client;
     trie_sanity_check.check_epochs(client);
     let height_after_resharding = latest_block_height.get();
 
-    // Delete `temporary_account`.
-    delete_account(&mut env, &rpc_id, &temporary_account, &rpc_id);
-    // Wait for garbage collection to kick in.
-    env.test_loop
-        .run_for(Duration::seconds((DEFAULT_GC_NUM_EPOCHS_TO_KEEP * params.epoch_length) as i64));
-    // Check that the deleted account is still accessible at archival node, but not at a regular node.
-    check_deleted_account_availability(
-        &mut env,
-        &archival_id,
-        &rpc_id,
-        temporary_account,
-        height_after_resharding,
-    );
+    if let Some(archival_id) = archival_id {
+        // Delete `temporary_account`.
+        delete_account(&mut env, &client_account_id, &temporary_account, &client_account_id);
+        // Wait for garbage collection to kick in.
+        env.test_loop.run_for(Duration::seconds(
+            (DEFAULT_GC_NUM_EPOCHS_TO_KEEP * params.epoch_length) as i64,
+        ));
+        // Check that the deleted account is still accessible at archival node, but not at a regular node.
```

Comment:
We want to test garbage collection even if no archival node is available.
For example, to trigger negative refcounts with delayed receipts.

Reply:
I see, then I'll just skip the call to the archival node